sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -865,3 +865,24 @@ def test_partial_dependence_bunch_values_deprecated():
 
     # "values" and "grid_values" are the same object
     assert values is grid_values
+
+
+def test_mixed_type_categorical():
+    """Check that we raise a proper error when a column has mixed types and
+    the sorting of `np.unique` will fail."""
+    X = np.array(["A", "B", "C", np.nan], dtype=object).reshape(-1, 1)
+
+    percentiles, is_categorical, grid_resolution = (
+        (0.05, 0.95),
+        [
+            True,
+        ],
+        30,
+    )
+    with pytest.raises(ValueError, match="The column #0 contains mixed data type"):
+        _grid_from_X(
+            X,
+            percentiles=percentiles,
+            is_categorical=is_categorical,
+            grid_resolution=grid_resolution,
+        )
```

Comment:
I'd rather check the error raised when calling the public API.

Reply:
done.